documentation.md

```diff
@@ -0,0 +1,205 @@
+
+1. Sign up at [3dsecure.io](https://www.3dsecure.io), this is out of scope of this document.
+2. Read the [getting started](#getting-started) guide for getting an
+   understanding of the authentication flow.
+3. Look at the [examples](#examples) in this documentation.
+4. Consult the [reference](#reference) to get details about the individual API endpoints.
+
+## Read me first
+
+<!---
+{{{
+-->
+
+This is the documentation for Clearhaus EMVCo 3-D Secure Server implementation.
+Our implementation is a SAAS implementation, offering a language-agnostic HTTP
+API integration.
+Briefly, a 3-D Secure Server is used in the financial industry, to:
+- **Facilitate cardholder _authentication_ in preparation for e.g. making a purchase.**
+
+The specification is developed by EMVCo, best described by [their own
+words](https://www.emvco.com/about/overview/):
+
+> EMVCo exists to facilitate worldwide interoperability and acceptance of
+> secure payment transactions
+
+The specification is public and can be found at https://www.emvco.com/emv-technologies/3d-secure/.
+This service supports all active versions of 3-D Secure version 2:
+
+- 2.1.0
+- 2.2.0
+
+During integration, you will need to refer to specification for _both_
+versions.  The EMVCo 3-D Secure has a different program name for each card
+scheme, this service supports the following card schemes:
+
+
+| Card Scheme     | Program name                                                 |
+| --------------- | ------------------------------------------------------------ |
+| Mastercard      | Mastercard Identity Check (previously Mastercard SecureCode) |
+| Visa            | Visa Secure (previously Verified by Visa)                    |
+
+<!---
+}}}
+-->
+
+## Getting Started
+
+### What is an authentication?
+
+An authentication in 3-D Secure is the process of verifying cardholder
+involvement in e.g. a purchase. An authentication results from and
+_authentication flow_ and proof of authentication is an
+_authentication value_.
+
+This figure illustrates the simplified authentication flow from a requestor
+point of view:
+
+<div class="img">
+  <img src="authentication.svg" alt="Authentication"/>
+</div>
+
+An _authentication flow_ is one of the following categories:
+* _Frictionless flow_
+
+   This flow includes messages 1-4, where the `authenticationValue`, `eci` and
+   `dsTransID` are returned in 4.
+
+* _Challenge flow_
+
+   This includes 1-8, where the `authenticationValue`, `eci` and `dsTransID`
+   are returned in 8.
+
+* _Out-of-band (OOB)_
+
+   This includes 1-8, where the `authenticationValue`, `eci` and `dsTransID`
+   are returned in 8.
+
+* _Decoupled Authentication_
+
+   This includes 1-4,6-8, where the `authenticationValue`, `eci` and `dsTransID`
+   are returned in 8.
+
+### Performing an authentication flow
+
+1. Use the [`/preauth`](#the-preauth-call) call to determine 3-D Secure version.
+2. Perform an authentication using the [`/auth`](#the-auth-call) call.
```

Comment:
That's a fair point. You do mean the endpoints in the API and not in the documentation just, correct?

Reply:
Thinking in terms of HATEOS, these two (the name, and the URL) are two different things, and then I'll care about the name (not the URL) because that is the thing we refer to and document. The URL is "just" the discoverable thing you use as a parameter (together with other parameters like the headers and the body) for your next request 🙂

Reply:
You're going to have to dumb it down for me :)

Reply:
Let's IRL it 🙂